src/main/java/com/facebook/openwifirrm/modules/ConfigManager.java

```diff
@@ -201,11 +228,12 @@ private void runImpl() {
 			for (ConfigListener listener : configListeners.values()) {
 				listener.receiveDeviceConfig(device.serialNumber, data.config);
 			}
-
-			// Check event flag
+			// Check if there are requested updates for this zone
+			// And if so, remove this zone from the set of to-be-updated zones
+			boolean isEvent = zonesToUpdateCopy.remove(device.venue);
```

Comment:
- I don't understand why you are removing the zone here?? What about the rest of the devices in the zone?
- You cannot use `device.venue`, you need to look up the zone from `deviceDataManager.getDeviceZone(device.serialNumber)`.